src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import org.thymeleaf.context.Context;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * This class performs various actions with jobs for the SLURM engine.
+ */
+@Service
+@Slf4j
+@ConditionalOnProperty(name = "grid.engine.type", havingValue = "SLURM")
+public class SlurmJobProvider implements JobProvider {
+    private static final String JOB_FILTER = "filter";
+    private static final String SQUEUE_COMMAND = "squeue";
+
+    /**
+     * The MapStruct mapping mechanism used.
+     */
+    private final SlurmJobMapper jobMapper;
+
+    /**
+     * The command execution mechanism used.
+     */
+    private final SimpleCmdExecutor simpleCmdExecutor;
+
+    private final int fieldsCount;
+
+    /**
+     * An object that forms the structure of an executable command according to a template.
+     */
+    private final GridEngineCommandCompiler commandCompiler;
+
+    public SlurmJobProvider(final SlurmJobMapper jobMapper,
+                            final SimpleCmdExecutor simpleCmdExecutor,
+                            final GridEngineCommandCompiler commandCompiler,
+                            @Value("${slurm.job.output-fields-count:52}") final int fieldsCount) {
+        this.jobMapper = jobMapper;
+        this.simpleCmdExecutor = simpleCmdExecutor;
+        this.commandCompiler = commandCompiler;
+        this.fieldsCount = fieldsCount;
+    }
+
+    @Override
+    public EngineType getProviderType() {
+        return EngineType.SLURM;
+    }
+
+    @Override
+    public Listing<Job> filterJobs(final JobFilter jobFilter) {
+        final CommandResult result = simpleCmdExecutor.execute(makeSqueueCommand(jobFilter));
+        if (result.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(result);
+        } else if (!result.getStdErr().isEmpty()) {
+            log.warn(result.getStdErr().toString());
+        }
+        return mapToJobListing(result.getStdOut());
+    }
+
+    /**
+     * Creates the structure of an executable command based on the passed filter.
+     *
+     * @param filter User-defined filter for selecting job data.
+     * @return The structure of an executable command.
+     */
+    private String[] makeSqueueCommand(final JobFilter filter) {
+        final Context context = new Context();
+        context.setVariable(JOB_FILTER, filter);
+        return commandCompiler.compileCommand(getProviderType(), SQUEUE_COMMAND, context);
+    }
+
+    private Listing<Job> mapToJobListing(final List<String> stdOut) {
+        if (stdOut.size() > 1) {
+            return new Listing<>(stdOut.stream()
+                    .skip(1)
+                    .map(jobDataList -> SacctCommandParser.parseSlurmJob(jobDataList, fieldsCount))
+                    .filter(parsedJobList -> !CollectionUtils.isEmpty(parsedJobList))
```

Comment:
We have Apache `ColectionUtils` as the project dependency. Since it has richer capabilities and already here - let's use `CollectionUtils.isNotEmpty` method